Symfony 5/PHP 8 compatibility

GeneraleCauchemar opened the pull request:
- Makes the bundle compatible with Symfony >= 5 & PHP >= 8, as well as Doctrine
ORM 2.13
- Adds typing to properties and methods
- Adds a CHANGELOG

GeneraleCauchemar committed:
Symfony 5/PHP 8 compatibility

- Makes the bundle compatible with Symfony >= 5 & PHP >= 8, as well as Doctrine
ORM 2.13
- Adds typing to properties and methods
- Adds a CHANGELOG

GeneraleCauchemar committed:
Merge branch 'master' of github.com:GeneraleCauchemar/doctrine-singleton-bundle

DjLeChuck commented on CHANGELOG.md:
```diff
@@ -0,0 +1,26 @@
+
+## [Unreleased]
+
+## [2.0.0] - 2023-01-10
```
Le `2.0.0` ne pointe vers rien dans le fichier, donc il ne faut pas l'entourer de `[]`.
Et d'ailleurs tu ne devrais pas le mettre du tout, tes modifications doivent résider dans `[Unreleased]` et c'est au moment de la création de la version que le mainteneur viendra également modifier le fichier CHANGELOG pour ajouter le `2.0.0`

GeneraleCauchemar replied:
C'est corrigé

DjLeChuck commented on DependencyInjection/Configuration.php:
```diff
@@ -8,7 +8,7 @@
 /**
```
Tout ce fichier est inutile puisque le bundle ne propulse pas de configuration

GeneraleCauchemar replied:
Fichier supprimé

DjLeChuck commented on DependencyInjection/UmanitDoctrineSingletonExtension.php:
```diff
@@ -21,29 +21,23 @@ class UmanitDoctrineSingletonExtension extends Extension implements PrependExten
         // Set configuration into params
         $rootName = 'umanit_doctrine_singleton';
         $container->setParameter($rootName, $config);
         $this->setConfigAsParameters($container, $config, $rootName);
```
Du coup, comme le fichier de configuration est à supprimer, pas mal de nettoyage à faire ici

GeneraleCauchemar replied:
J'ai mis à jour, j'espère avoir bien compris ce que je devais laisser, j'ai toujours un doute là :D

DjLeChuck replied:
C'était bien ça, mais du coup tu peux également supprimer la méthode `private function setConfigAsParameters(ContainerBuilder &$container, array $params, $parent)` qui ne sert plus.

GeneraleCauchemar replied:
Fait :)

DjLeChuck commented on Doctrine/EventSubscriber/SingletonSubscriber.php:
```diff
@@ -42,19 +37,20 @@ public function getSubscribedEvents()
             // Event to add filters
             $event = new FilterSingletonEvent($entity, $filters);
-            $this->eventDispatcher->dispatch(FilterSingletonEvent::SINGLETON_FILTER_EVENT, $event);
+            $this->eventDispatcher->dispatch($event, FilterSingletonEvent::SINGLETON_FILTER_EVENT);
```
Quitte à tout mettre à jour, autant corriger cet appel.

Les appel à dispatch ne prennent plus qu'un seul argument maintenant (l'event) et non 2. Tu dois le faire dans ta debug bar Symfo normalement.

GeneraleCauchemar replied:
Vu ensemble, la 5.4 permet en deuxième argument (facultatif) le nom de l'event

DjLeChuck commented on Event/FilterSingletonEvent.php:
```diff
@@ -12,72 +12,35 @@ class FilterSingletonEvent extends Event
 {
     const SINGLETON_FILTER_EVENT = 'umanit_doctrine_singleton.filter';
```
Du coup cette constante ne servira plus, l'écoute d'un événement dans un subscriber ne se faisant plus via `MonEvent::MA_CONSTANTE`, mais sur `MonEvent::class` (donc son FQCN)

GeneraleCauchemar replied:
Voir commentaire ci-dessus

GeneraleCauchemar committed:
Changes requested on PR

- Modifies the CHANGELOG file
- Removes useless Configuration file
- Updates extension file to remove references to deleted config

GeneraleCauchemar committed:
Changes requested on PR

- Removes redundant setConfigAsParameters method from extension file
